src/StreamJsonRpc.Tests/JsonRpcClient20InteropTests.cs

```diff
@@ -149,6 +150,147 @@ public async Task InvokeWithParameterPassedAsObjectAsync_NoParameter()
+        Assert.Equal(sum2, sum);
+
+        this.Send(new
+        {
+            jsonrpc = "2.0",
+            id = request["id"].Value<long>(),
+            result = sum,
+        });
+
+        int result = await invokeTask;
+        Assert.Equal(sum2, result);
+    }
+
+    [Fact]
+    public async Task InvokeWithProgressParameter()
+    {
+        int sum = 0;
+        ProgressWithCompletion<int> progress = new ProgressWithCompletion<int>(report =>
+        {
+            sum += report;
+        });
+
+        int n = 3;
+        Task<int> invokeTask = this.clientRpc.InvokeWithParameterObjectAsync<int>("test", new { Bar = n, Progress = progress });
+
+        JToken request = await this.ReceiveAsync();
+
+        long progressID = request["params"]["Progress"].Value<long>();
+
+        // Send responses as $/progress
+        int sum2 = 0;
+        for (int i = 0; i < n; i++)
+        {
+            string content = "{ \"jsonrpc\": \"2.0\", \"method\": \"$/progress\", \"params\": { \"token\": " + progressID + ", \"value\": " + i + ", } ,}";
+            JObject json = JObject.Parse(content);
+
+            this.Send(json);
+
+            sum2 += i;
+        }
+
+        System.Threading.Thread.Sleep(1000);
+        Assert.Equal(sum2, sum);
+
+        this.Send(new
+        {
+            jsonrpc = "2.0",
+            id = request["id"].Value<long>(),
+            result = sum,
+        });
+
+        int result = await invokeTask;
+        Assert.Equal(sum2, result);
+    }
+
+    [Fact]
+    public async Task InvokeWithProgressParameter_NoMemoryLeakConfirm()
+    {
+        WeakReference weakRef = await this.InvokeWithProgressParameter_NoMemoryLeakConfirm_Helper();
+        GC.Collect();
+        Assert.False(weakRef.IsAlive);
+    }
+
+    [Fact]
+    public async Task InvokeWithProgressParameterAsObject_CheckProgressObjectIsReplacedForId()
+    {
+        ProgressWithCompletion<int> progress = new ProgressWithCompletion<int>(report => { });
+
+        Task<int> invokeTask = this.clientRpc.InvokeWithParameterObjectAsync<int>("test", new { Bar = "value", Progress = progress });
+
+        JToken request = await this.ReceiveAsync();
+
+        Assert.Equal(JTokenType.Object, request["params"].Type);
+        Assert.Equal("value", request["params"]["Bar"].ToString());
+        Assert.Equal(JTokenType.Integer, request["params"]["Progress"].Type);
+    }
+
+    [Fact]
+    public async Task InvokeWithProgressParameterAsArray_CheckProgressObjectIsReplacedForId()
+    {
+        ProgressWithCompletion<int> progress = new ProgressWithCompletion<int>(report => { });
+
+        Task<int> invokeTask = this.clientRpc.InvokeAsync<int>("test", new object[] { "value", progress });
+
+        JToken request = await this.ReceiveAsync();
+
+        Assert.Equal(JTokenType.Array, request["params"].Type);
+        Assert.Equal("value", request["params"][0].ToString());
+        Assert.Equal(JTokenType.Integer, request["params"][1].Type);
+    }
+
```

Comment:
I don't know what this test covers (aside from implementation detail) that your prior tests don't already confirm (i.e. that it works). I suggest you remove these tests unless I'm missing something interesting about them.